src/python_testing/TC_TCTL_2_3.py

```diff
@@ -0,0 +1,98 @@
+#
+#    Copyright (c) 2024 Project CHIP Authors
+#    All rights reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+# === BEGIN CI TEST ARGUMENTS ===
+# test-runner-runs:
+#   run1:
+#     app: ${ALL_CLUSTERS_APP}
+#     app-args: --discriminator 1234 --KVS kvs1 --trace-to json:${TRACE_APP}.json
+#     script-args: >
+#       --storage-path admin_storage.json
+#       --commissioning-method on-network
+#       --discriminator 1234
+#       --passcode 20202021
+#       --endpoint 1
+#       --trace-to json:${TRACE_TEST_JSON}.json
+#       --trace-to perfetto:${TRACE_TEST_PERFETTO}.perfetto
+#       --PICS src/app/tests/suites/certification/ci-pics-values
+#     factory-reset: true
+#     quiet: true
+# === END CI TEST ARGUMENTS ===
+
+import chip.clusters as Clusters
+from chip.testing.matter_testing import MatterBaseTest, TestStep, default_matter_test_main, has_feature, run_if_endpoint_matches
+from mobly import asserts
+
+
+class TC_TCTL_2_3(MatterBaseTest):
+    def desc_TC_TCTL_2_3(self) -> str:
+        return "[TC-TCTL-2.3] Optional temperature level attributes with DUT as Server"
+
+    def pics_TC_TCTL_2_3(self):
+        """Return the PICS definitions associated with this test."""
+        pics = [
+            "TCTL.S",      # Temperature Control as a Server
+            "TCTL.S.F01",  # Does a device support temperature level feature
+        ]
+        return pics
+
+    def steps_TC_TCTL_2_3(self) -> list[TestStep]:
+        steps = [
+            TestStep(1, "Commissioning, already done", is_commissioning=True),
+            TestStep(2, "TH reads from the DUT the SelectedTemperatureLevel attribute"),
+            TestStep(3, "TH reads from the DUT the SupportedTemperatureLevels attribute and verifies string lengths"),
```

Comment:
Would you mind adding in the expected outcome strings here too? It's not a huge deal, but it makes it much easier to update the test plan.

Reply:
Sure, added